### PR TITLE
Support responsive/adaptive design with viewport size

### LINKED ISSUE #123: Support responsive design with media query

### PR DESCRIPTION
**Pull request by wwwillchen**
Fixes #123

### CODE REVIEW

File: mesop/server/server.py [hunk 1]
```diff
@@ -132,7 +133,17 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
         else:
           yield from render_loop(path=ui_request.path, init_request=True)
       elif ui_request.HasField("user_event"):
-        runtime().context().update_state(ui_request.user_event.states)
+        event = ui_request.user_event
+        if event.HasField("navigation") or event.HasField("resize"):
+          if event.HasField("resize"):
+            runtime().context().set_viewport_size(event.resize.viewport_size)
+          elif event.HasField("navigation") and event.navigation.HasField(
```
Comment (richard-to) [inline mesop/server/server.py:140]: [question] What are the cases where viewport_size is not included with the navigation event?

**Review by richard-to — APPROVED** [review]
Awesome. Definitely a much needed feature.

File: docs/api/viewport_size.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+```py
+import mesop as me
+
+@me.page()
+def page():
+    if me.viewport_size().width > 640:
+        width = me.viewport_size().width / 2
+    else:
+        width = me.viewport_size().width
+    me.box(style=me.Style(width=width))
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply docs/api/viewport_size.md:23]: hah, that's fair.

File: mesop/server/server.py (hunk 1 above)
Comment (wwwillchen) [reply mesop/server/server.py:140]: viewport_size isn't included in a regular navigation (me.navigate), but it's included in navigations triggered by a hot reload.

I've treated hot reload as a type of navigation b/c it uses (mostly) the same code paths, but it makes this a bit more confusing. I can refactor this later.

**Pull request merged**